File: avalon-support/avalon-oauth2-support/avalon-oauth2-authorization-server/src/main/java/io/github/lishangbu/avalon/oauth2/authorizationserver/authentication/SmsAuthenticationToken.java
package io.github.lishangbu.avalon.oauth2.authorizationserver.authentication;

import java.util.Collection;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

/// 短信验证码登录的认证令牌
///
/// 用于承载手机号与短信验证码，并与 AuthenticationManager 协作完成认证
///
/// @author lishangbu
/// @since 2026/3/13
public class SmsAuthenticationToken extends UsernamePasswordAuthenticationToken {

    public SmsAuthenticationToken(String phoneNumber, String smsCode) {
        super(phoneNumber, smsCode);
    }

    public SmsAuthenticationToken(
            Object principal,
            Object credentials,
            Collection<? extends GrantedAuthority> authorities) {
        super(principal, credentials, authorities);
    }
}

File: avalon-support/avalon-oauth2-support/avalon-oauth2-authorization-server/src/main/kotlin/io/github/lishangbu/avalon/oauth2/authorizationserver/authentication/SmsAuthenticationToken.kt
package io.github.lishangbu.avalon.oauth2.authorizationserver.authentication

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.GrantedAuthority

/**
 * 短信验证码登录的认证令牌 用于承载手机号与短信验证码，并与 AuthenticationManager 协作完成认证
 *
 * @author lishangbu
 * @since 2026/3/13
 */
class SmsAuthenticationToken : UsernamePasswordAuthenticationToken {
    constructor(phoneNumber: String, smsCode: String) : super(phoneNumber, smsCode)

    constructor(
        principal: Any?,
        credentials: Any?,
        authorities: Collection<GrantedAuthority>,
    ) : super(principal ?: "", credentials ?: "", authorities)
}
